src/modules/common/components/customRoute/index.js

```diff
@@ -19,9 +20,15 @@ const CustomRoute = ({ path, exact, isPrivate, forbiddenTokens, component, t, hi
   const { isMigrated } = useCheckLegacyAccount(currentAccount?.metadata?.pubkey);
   const { search = '' } = history.location;
   const { accounts } = useAccounts();
+  const { mainChainNetwork } = useSettings('mainChainNetwork');
 
   Piwik.tracking(history, token);
 
+  if (!mainChainNetwork && path !== routes.selectNetwork.path) {
+    history.replace({ pathname: routes.selectNetwork.path });
+    return <div />;
```

Comment:
is the div here to prevent flicker?
Why cant we just use `<Redirect` ?

Reply:
resolved